src/edit.rs

```diff
@@ -128,7 +128,8 @@ impl<'out, 'prompt, H: Helper> State<'out, 'prompt, H> {
         if self.layout.cursor == cursor {
             return Ok(());
         }
-        if self.highlight_char() {
+        if self.highlight_char() || self.has_hint() {
```

Comment:
Why moving the cursor should invalidate the hint ?
At least, this should be configurable, not hardcoded.

Reply:
In my example in the initial description go the pull request the cursor is moved in steps (3) to (7). In all these positions it does not make sense to insert the remaining partial part of the hint. Try to use it with the app I uploaded https://github.com/jerbs/fsidx. It has a shell mode with hints and completions implemented for long options starting with "--". I don't see in which scenario the currently implemented behaviour is useful.